edx_event_bus_redis/internal/consumer.py

```diff
@@ -205,88 +214,36 @@ def _consume_indefinitely(self):
                     consecutive_errors += 1
                     self.record_event_consuming_error(run_context, e, msg)
                     # Kill the infinite loop if the error is fatal for the consumer
-                    _, redis_error = self._get_redis_message_and_error(message=msg, error=e)
-                    # https://redis.readthedocs.io/en/stable/exceptions.html#redis.exceptions.TryAgainError
-                    if redis_error:  # and redis_error.fatal()
+                    fatal = self._is_fatal_redis_error(error=e)
+                    if fatal:
                         raise e
-                    # Prevent fast error-looping when no event received from broker. Because
-                    # DeserializingConsumer raises rather than returning a Message when it has an
-                    # error() value, this may be triggered even when a Message *was* returned,
-                    # slowing down the queue. This is probably close enough, though.
-                    if msg is None:
+                    # Prevent fast error-looping when no event received from broker.
+                    if redis_raw_msg is None:
                         time.sleep(POLL_FAILURE_SLEEP)
-                # if msg:
-                    # theoretically we could just call consumer.commit() without passing the specific message
-                    # to commit all this consumer's current offset across all partitions since we only process one
-                    # message at a time, but limit it to just the offset/partition of the specified message
-                    # to be super safe
-                    # self.consumer.commit(message=msg)
+                if msg and msg.msg_id:
+                    self.consumer.ack(msg.msg_id)
```

Comment:
@rgraber Hmm, the consumer will exit with an exception. 
The consumer can then be started again with the msg_id that was ack'd using `--last_read_msg_id` command line argument.

What would be the best way to handle this kind of situation?

Reply:
To clarify, the situation I'm more concerned with is something like a kubernetes restart:
1. we ack the message id
2. Kubernetes (or whatever) restarts the process running this consumer
3. The message has been acked, but the id on the consumer is the id before the last message. 

Will this cause a problem or will it just move on? Or re-read the ack'd message?

Reply:
@navinkarkera I believe there's an example of how to handle this kind of recovery without explicitly passing in a msg id here: https://redis.io/docs/data-types/streams-tutorial/#creating-a-consumer-group which I think solves for this case. There's probably still discussion to be had about implementing autoclaim in the case of completely dead consumers, though I think that's less relevant right now.

Reply:
@bmtcril Yes, we don't need to pass ids explicitly for unread as well as unacknowledged messages. Only when we want to replay events from history, `set_id` comes into play. I have updated arguments for the command, taking in `check_backlog` which is used for processing unread messages (any messages which were not read by any consumer under the passed group) and `last_read_msg_id` which allows us to set the read msg id, so the consumer will process messages after the provided msg. @rgraber 

You can test these arguments by publishing some messages using `make produce_test_event` before starting any consumer. Next start the consumer with `--check_backlog` flag, it will process the messages that you published before starting the consumer. Now start the consumer with `--last_read_msg_id <any msg id>` and it will process all messages after the passed id.

Reply:
I haven't read this carefully, but Kafka has at-least-once processing guarantees that can be made. I'm not sure if Redis can make the same guarantee, and what is needed to achieve that. Good luck in working this out.

Also note that we added audit logging to our Kafka implementation for now. I don't know when and if it will get too noisy, but it could help for investigating any odd behaviors.